tests/jsmarionette/runner/marionette-js-runner/host/session.js

```diff
@@ -31,10 +31,21 @@ Session.prototype = {
 
     return this.host.request('/get_crash_info', request)
     .then(function(info) {
+      // If info is empty, we didn't crash the process. We threw an error
+      // in JavaScript code instead.
+      if (Object.keys(info).length === 0) {
+        // The error we were passed originally will suffice in this case.
+        return err;
```

Comment:
Unfortunately the way we parse the crash info, it needs to return a value always. So we need to check upstream to see if the 'info' returned from the python runner contains anything. In the case where the process didn't crash, this is always an empty object.

In the case of a real crash we'll still have something in err, just different data.